dawn/renderer/components/Editor.js

```diff
@@ -134,48 +155,90 @@ class Editor extends React.Component {
     setTimeout(() => this.refs.CodeEditor.editor.resize(), 0.1);
   }
 
-  sendCode(command) {
+  upload() {
+    const filepath = this.props.filepath;
+    if (filepath == null) {
+      dialog.showMessageBox({
+        type: 'warning',
+        buttons: ['Close'],
+        title: 'No Files',
+        message: 'No file? No upload',
+      });
+      console.log('No file? No upload');
+      return;
+    }
     const correctedText = this.correctText(this.props.editorCode);
     if (correctedText !== this.props.editorCode) {
       this.props.onAlertAdd(
         'Invalid characters detected',
         'Your code has non-ASCII characters, which won\'t work on the robot. ' +
-        'Please remove them and try again.'
+        'Please remove them and try again.',
       );
-      return false;
+      return;
     }
-    console.log('Deprecated: sendCode');
-    console.log(`Debug Info: ${command} EOD`);
-    return true;
-  }
-
-  upload() {
-    this.sendCode('upload');
+    const conn = new Client();
```

Comment:
Issue with moving into a saga is the amount of variables needed to be transmitted between processes. To be honest, these are also things that should be waited on.